### DISCUSSION TITLE
Dockerfile to build ARM-64 image for rabbitmq-server

### DISCUSSION THREAD

**Discussion by git-abhijit** [docker-library/rabbitmq#697] [Q&A] [answered]
Hello  Forum, 

Need one help I am trying to build ARM64 and AMD64 based rabbitmq-server ; version 3.10 ; images on ubuntu OS using docker buildx tool. Can the docker file --> https://github.com/docker-library/rabbitmq/blob/92cf3ab7107d78127192516c4011d2daef036700/3.10/ubuntu/Dockerfile  ; be used to built for both architectures ? 

Appreciate help on this. Thanks. 

Thanks and Regards
Abhijit

**Comment by LaurentGoderre** [comment]
It already has been built for both of these architectures: https://hub.docker.com/layers/library/rabbitmq/3.10.25-management/images/sha256-a9de5724d2ebc1322d0ad894d5a82226c27c6cf1d957949da7fb0ee9319e5c68?context=explore

**Reply by git-abhijit** [reply]
Thanks Laurent for the detail. Yes it's built for both architectures. 

Actually my requirement is to build locally for both the architecutres using the docker file shared earlier. I will be using docker buildx  to build images for both architectures on a AMD64 VM. 

Then the  Dockerfile which I had shared earlier should work to build images for both architectures , correct ?

**Reply by lukebakken** [reply]
> Then the Dockerfile which I had shared earlier should work to build images for both architectures , correct ?

Why don't you give it a try, yourself, rather than asking?

**Reply by git-abhijit** [reply]
:) Well I gave it a try facing some issues related to building OpenSSL  or Erlang/OTP trying to debug that now.

**Reply by LaurentGoderre** [reply]
It was successfully built 2 months ago but it's possible that something changed since then. Those were also built using buildx

**Reply by lukebakken** [reply]
Works on my machine (Ubuntu 22.04). Note that the commit I'm using is the parent to [this commit](https://github.com/docker-library/rabbitmq/commit/d450e17da7d68f9b4f152269adf54d96d66e055c) where 3.10 was removed from this project.

```
git checkout d450e17da7d68f9b4f152269adf54d96d66e055c~1
cd 3.10/ubuntu
docker build --tag docker-library_rabbitmq-gh-697-ubuntu .
```

Output:

[docker-build-3.10-ubuntu-849609c.txt](https://github.com/docker-library/rabbitmq/files/15179554/docker-build-3.10-ubuntu-849609c.txt)

**Reply by git-abhijit** [reply]
Failing for the ARM build.  Need guidance on this. 

Seems like the build process failed during the installation of OTP (Erlang/OTP) for an ARM64 architecture.  Do we need some specific pre-requisite packages related to ARM. 
[DockerFile_ARM_RabbitMQ.txt](https://github.com/docker-library/rabbitmq/files/15278070/DockerFile_ARM_RabbitMQ.txt)

**Reply by LaurentGoderre** [reply]
It's not finding libssl

```
301.8 strip: '/usr/local/lib/x86_64-linux-gnu/libssl.so.1.1': No such file
```

**Reply by tianon** [reply, marked as answer]
We can even `docker build` (or `docker buildx build`) directly from the commit referenced above (the parent of https://github.com/docker-library/rabbitmq/commit/d450e17da7d68f9b4f152269adf54d96d66e055c, which is https://github.com/docker-library/rabbitmq/commit/849609cdb8a5181e8e830de29030ebd2bdcb62e0):

`docker build --pull --platform linux/arm64 'https://github.com/docker-library/rabbitmq.git#849609cdb8a5181e8e830de29030ebd2bdcb62e0:3.10/ubuntu'`

**Reply by lukebakken** [reply]
@git-abhijit - please be sure to use the correct commit, as @tianon indicated in [this comment](https://github.com/docker-library/rabbitmq/discussions/697#discussioncomment-9388505)

**Reply by git-abhijit** [reply]
Thank you @LaurentGoderre @tianon @lukebakken 
The problem is resolved. Thank you all for the suggestions ; build was successful after taking the above commit and removing from Dockerfile  strip: /usr/local/lib/x86_64-linux-gnu/libssl.so.1.1

**Comment by lukebakken** [comment]
For what it's worth, RabbitMQ 3.10 is completely out of support.
